akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessKeepAlivePinger.scala

```diff
@@ -0,0 +1,292 @@
+        }
+        Behaviors.same
+      case SendKeepAliveDone(`currentRevision`) =>
+        timers.startSingleTimer(Tick, Tick(currentRevision), settings.keepAliveInterval)
+        Behaviors.same
+      case Pause(revision, replyTo) =>
+        context.log.debug2("Pausing sharded daemon process pinger [{}] (revision [{}])", daemonProcessName, revision)
+        if (revision <= currentRevision) {
+          timers.cancel(Tick)
+        }
+        replyTo ! StatusReply.Success(context.self.path)
+        paused(revision)
+
+      case SendKeepAliveDone(oldRevision) =>
+        context.log.debugN(
+          "KeepAlive for [{}] got SendKeepAliveDone from old revision [{}], current is [{}], ignoring",
+          daemonProcessName,
+          oldRevision,
+          currentRevision)
+        Behaviors.ignore
+
+      case Tick(oldRevision) =>
+        context.log.debugN(
+          "Keep alive for [{}] got tick from old revision [{}], current is [{}], ignoring",
+          daemonProcessName,
+          oldRevision,
+          currentRevision)
+        Behaviors.ignore
+      case Resume(`currentRevision`, _, replyTo) =>
+        context.log.debug2(
+          "Sharded daemon process pinger [{}] got start for already started revision (revision [{}]",
+          daemonProcessName,
+          currentRevision)
+        replyTo ! StatusReply.Success(context.self.path)
+        Behaviors.ignore
+
+      case Resume(otherRevision, _, replyTo) =>
+        context.log.debugN(
+          "Sharded daemon process pinger [{}] got start for unexpected revision (revision [{}], current [{}])",
+          daemonProcessName,
+          otherRevision,
+          currentRevision)
+        replyTo ! StatusReply.Success(context.self.path)
+        Behaviors.ignore
+
+      case unexpected =>
+        context.log.warn2(
+          "{}: Unexpected message to pinger in state 'start': {}",
+          daemonProcessName,
+          unexpected.getClass)
+        Behaviors.unhandled
+
+    }
+  }
+
+  /**
+   * Rescale in progress, don't ping workers until coordinator says so
+   */
+  private def paused(pausedRevision: Long): Behavior[Message] = Behaviors.receiveMessage {
+    case Pause(_, replyTo) =>
+      replyTo ! StatusReply.Success(context.self.path)
+      Behaviors.same
+    case Resume(revision, newNumberOfProcesses, replyTo) =>
+      if (revision >= pausedRevision) {
+        context.log.debug2("{}: Un-pausing sharded daemon process pinger (revision [{}])", daemonProcessName, revision)
+        replyTo ! StatusReply.Success(context.self.path)
+        context.self ! Tick(revision)
+        start(revision, newNumberOfProcesses)
+      } else {
+        context.log.warn2(
+          "{}, Paused sharded daemon process pinger, got unexpected start for old revision [{}], ignoring",
+          daemonProcessName,
+          revision)
+        replyTo ! StatusReply.Error("Old revision")
```

Comment:
include the revision numbers in the error message, for easier correlation on other side?

Reply:
Added in 5a544fc97cc1136365569407848facc355b5a088